packages/vc-verification/src/resolver/intermediate-data-cache.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  CachedDIDDocument,
+  CachedRoleCredential,
+  CachedRoleDefinition,
+  RoleEIP191JWT,
+} from '../models';
+import { IDataCache } from './idata-cache';
+import { IServiceEndpoint } from '@ew-did-registry/did-resolver-interface';
+import { IRoleDefinitionV2 } from '@energyweb/credential-governance';
+import { VerifiableCredential } from '@ew-did-registry/credentials-interface';
+import type { RoleCredentialSubject } from '@energyweb/credential-governance';
+
+/**
+ * A class to aggreagate data from EVMs
+ */
+export class IntermediateDataCache implements IDataCache {
```

Comment:
IMO, the `IDataCache` interface is too large and introduces Interface Segregation Principle violations. I recommend splitting it into several smaller interfaces 
```suggestion
export class IntermediateDataCache implements IRoleCache, IDIDDocCache, IRoleCredentialCache, IClearableCache {
```

Reply:
I would also consider having separate caches per object type. IMO, this would better follow the SRP and it seems the clients of the cache only needs a single cache type. E.g. `EthersProviderIssuerResolver` only needs role definitions.
So could do:
```
RoleDefinitionCache implements IRoleDefinitionCache;
DIDDocCache implemented IDIDDocCache;
RoleCredentialCache implements IRoleCredentialCache;
```